Added scroller to trait page and changed sample tables to use it

zsloan opened the pull request:
## Description
This PR will add Scroller functionality to some GN tables and address the various form submissions/javascript issues caused by it. Since all table rows (for example sample values in the show_trait page) won't be loaded in the DOM at once anymore (since Scroller loads them as you scroll), the way data is fetched/change will need to be changed to use DataTables' API.

#### How should this be tested?
Load a trait page (or any other page) with a large table, and it should load faster. For example - http://genenetwork.org/show_trait?trait_id=10007&dataset=HSNIH-PalmerPublish

#### Any background context you want to provide?
We've had problems for a while with large tables taking a long time to load. I looked at Scroller once in the past, but I didn't realize at the time that there was a way to address the issue of all rows not being directly accessible by javascript/form submission.

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots (if appropriate)
N/A

#### Questions
N/A

zsloan committed:
Added scroller to trait page and changed sample tables to use it

* wqflask/wqflask/templates/show_trait.html - Imported scroller (from
the Google cdn temporarily for testing) and activated it for the sample
tables, as well as disabled options that conflicted with it (like
paging: false)

BonfaceKilz commented on wqflask/wqflask/templates/show_trait.html:
```diff
@@ -352,16 +353,15 @@ <h3 class="panel-title">
+                  "scrollY": "50vh",
+                  'scroller':  true,
+                  'scrollCollapse': true
+                  // 'processing': true,
+                  // 'language': {
+                  //   'loadingRecords': '&nbsp;',
+                  //   'processing': 'Loading...'
+                  // }
```
**suggestion: nitpick:** It's a good idea to remove unnecessary comments. This class of comments--commented out code-- after a while becomes "dead code" that either usually ends either being clutter or outright misleading.

BonfaceKilz commented on wqflask/wqflask/templates/show_trait.html:
```diff
@@ -502,16 +502,15 @@ <h3 class="panel-title">
-                    'loadingRecords': '&nbsp;',
-                    'processing': 'Loading...'
-                  }
+                  "scrollY": "50vh",
+                  'scroller':  true,
+                  'scrollCollapse': true
+                  // 'processing': true,
+                  // 'language': {
+                  //   'loadingRecords': '&nbsp;',
+                  //   'processing': 'Loading...'
+                  // }
```
See above.

zsloan commented:
Ah, I wasn't planning on keeping that there, but I guess there's no reason
to not delete it since the old code is still accessible.

On Tue, Sep 15, 2020 at 5:52 PM BonfaceKilz <notifications@github.com>
wrote:

> *@BonfaceKilz* commented on this pull request.
> ------------------------------
>
> In wqflask/wqflask/templates/show_trait.html
> <https://github.com/genenetwork/genenetwork2/pull/442#discussion_r489048143>
> :
>
> > +                  // 'processing': true,
> +                  // 'language': {
> +                  //   'loadingRecords': '&nbsp;',
> +                  //   'processing': 'Loading...'
> +                  // }
>
> *suggestion: nitpick:* It's a good idea to remove unnecessary comments.
> This class of comments--commented out code-- after a while becomes "dead
> code" that either usually ends either being clutter or outright misleading.
> ------------------------------
>
> In wqflask/wqflask/templates/show_trait.html
> <https://github.com/genenetwork/genenetwork2/pull/442#discussion_r489048286>
> :
>
> > +                  "scrollY": "50vh",
> +                  'scroller':  true,
> +                  'scrollCollapse': true
> +                  // 'processing': true,
> +                  // 'language': {
> +                  //   'loadingRecords': '&nbsp;',
> +                  //   'processing': 'Loading...'
> +                  // }
>
> See above.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/442#pullrequestreview-489160631>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGDQGSRABIWCS7IDNI3SF7V3TANCNFSM4RNPZ56A>
> .
>

BonfaceKilz commented:
zsloan <notifications@github.com> writes:

> Ah, I wasn't planning on keeping that there, but I guess there's no reason
> to not delete it since the old code is still accessible.

I normally try to get into the habit of cleaning
things up as soon as I can, because I forget :)

> On Tue, Sep 15, 2020 at 5:52 PM BonfaceKilz <notifications@github.com>
> wrote:
>
>> *@BonfaceKilz* commented on this pull request.
>> ------------------------------
>>
>> In wqflask/wqflask/templates/show_trait.html
>> <https://github.com/genenetwork/genenetwork2/pull/442#discussion_r489048143>
>> :
>>
>> > + // 'processing': true,
>> + // 'language': {
>> + // 'loadingRecords': '&nbsp;',
>> + // 'processing': 'Loading...'
>> + // }
>>
>> *suggestion: nitpick:* It's a good idea to remove unnecessary comments.
>> This class of comments--commented out code-- after a while becomes "dead
>> code" that either usually ends either being clutter or outright misleading.
>> ------------------------------
>>
>> In wqflask/wqflask/templates/show_trait.html
>> <https://github.com/genenetwork/genenetwork2/pull/442#discussion_r489048286>
>> :
>>
>> > + "scrollY": "50vh",
>> + 'scroller': true,
>> + 'scrollCollapse': true
>> + // 'processing': true,
>> + // 'language': {
>> + // 'loadingRecords': '&nbsp;',
>> + // 'processing': 'Loading...'
>> + // }
>>
>> See above.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/genenetwork/genenetwork2/pull/442#pullrequestreview-489160631>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AANQJGDQGSRABIWCS7IDNI3SF7V3TANCNFSM4RNPZ56A>
>> .
>>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.*

-- 
Bonface M. K. (https://www.bonfacemunyoki.com)
Chief Emacs Mchochezi
GPG key = D4F09EB110177E03C28E2FE1F5BBAE1E0392253F

zsloan committed:
Applied PEP8 and removed commented out code

zsloan committed:
Changed se_exists to a property so the SampleList instance can include
that when converted to a dict (which is needed to use it in the
javascript)

* wqflask/wqflask/show_trait/SampleList.py - Directly set se_exists as a
property of SampleList instead of requiring a method be called; I could
have set self.se_exists by calling the se_exists method, but that would
have resulted in the convoluted line "self.se_exists = self.se_exists()"
and the se_exists method only consisted of one line anyways

zsloan committed:
Changed references to se_exists to just get it as a property of
SampleList instead of calling the method

* wqflask/wqflask/show_trait/show_trait.py - see above
* wqflask/wqflask/templates/show_trait.html - see above + removed some
commented out code

zsloan commented:
After spending some time with this, one of the main issues seems to be making the DataTables instances accessible to the show_trait.js and show_trait_mapping_tools.js files. I think the easiest way to do this might be to put the event listeners in the template JS and have those pass the DataTables instances to the functions that do whatever is supposed to happen (updating figures, hiding rows, etc). Otherwise I think a bunch of existing template logic would need to be converted to JS and various template variables would need to be made accessible to the JS code (which I think has a cost since it's basically adding to what has to be loaded by the browser).

BonfaceKilz commented:
zsloan <notifications@github.com> writes:

> After spending some time with this, one of the main issues seems to be making the DataTables instances
> accessible to the show_trait.js and show_trait_mapping_tools.js files. I think the easiest way to do this
> might be to put the event listeners in the template JS and have those pass the DataTables instances to the
> functions that do whatever is supposed to happen (updating figures, hiding rows, etc). Otherwise I think a
> bunch of existing template logic would need to be converted to JS and various template variables would need to
> be made accessible to the JS code (which I think has a cost since it's basically adding to what has to be
> loaded by the browser).
>

Yeah. Go for something that works. Then worry
about the costs later.

> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.*

-- 
Bonface M. K. (https://www.bonfacemunyoki.com)
Chief Emacs Mchochezi
GPG key = D4F09EB110177E03C28E2FE1F5BBAE1E0392253F

zsloan committed:
Made a bunch of changes to move trait page DataTables initialization to
a separate file (initialize_show_trait_tables.js). The biggest
complication was getting the order of attributes in the rows to sync
with the order of atribute namees in the column headers. Previously this
logic was all in the template.
* wqflask/base/webqtlCaseData.py - added attribute first_attr_col as a
very awkward solution to passing the column position into the column
render function in situations where there are case attribute columns
(which can be variable in number)
* wqflask/wqflask/show_trait/show_trait.py - Replace "attribute_names"
in js_data with "attributes" (which allows the JS access to more
information) and also pass new se_exists and
has_num_cases variables with js_data, so the javascript has access to
whether or not those columns exist in the table
* wqflask/wqflask/static/new/javascript/show_trait.js - Change case
attribute-related logic to use js_data.attributes instead of
js_data.attribute_names
* wqflask/wqflask/templates/show_trait.html - Removed table
initialization from template
* wqflask/wqflask/static/new/javascript/initialize_show_trait_tables.js
- new file that initializes tables and reproduces what the template
logic used to do with JS logic

zsloan committed:
Made some style changes with whitespace/line-breaks

zsloan committed:
Changed JS on trait page that updates basic stats and graphs/figures to
use the DataTables API to pull sample values

* wqflask/wqflask/static/new/javascript/show_trait.js - Changed the way
sample name/val/var are pulled from the table to instead use DataTables'
API + removed an unused event listener
* wqflask/wqflask/static/new/javascript/initialize_show_trait_tables.js
- added "var" before a defined variable for consistency

zsloan commented:
Just as an update on this, I'm running into some problems related to the speed with which the DataTables API pulls values out of the cells of a large table. It's not a big deal if a table just has <500 rows, but for something like the HSNIH-Palmer data it's prohibitively slow any time a table value is changed (or a transform is run, etc).

I'm not sure how to address this (or if it even can be reasonably addressed without some very significant change to the way this works). I think that all of this data would need to be stored outside of the table and updated when specific rows/cells are changed, because I think that it's probably always going to be very slow to use the DataTables API to pull from all table rows. It would probably be faster if it was just pulling the table data, but because of the text inputs it has to actually pull the rendered cell contents.

zsloan commented:
I think I managed to solve the issue of speed. It actually seems significantly faster than it was before using Scroller now (for pulling data out of the table).

BonfaceKilz commented:
zsloan <notifications@github.com> writes:

> Just as an update on this, I'm running into some problems related to the speed with which the DataTables API
> pulls values out of the cells of a large table. It's not a big deal if a table just has <500 rows, but for
> something like the HSNIH-Palmer data it's prohibitively slow any time a table value is changed (or a transform
> is run, etc).
>
> I'm not sure how to address this (or if it even can be reasonably addressed without some very significant
> change to the way this works). I think that all of this data would need to be stored outside of the table and
> updated when specific rows/cells are changed, because I think it that it's probably always going to be very
> slow to use the DataTables API to pull from all table rows. It would probably be faster if it was just pulling
> the table data, but because of the text inputs it has to actually pull the rendered cell contents.
>

Perhaps using some web-storage API is of
consideration? Though IMO that would make things
really complicated. I'd leave this open, and first
prioritize merging Udit's work. We can figure
something out in due time.

> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.*

-- 
Bonface M. K. (https://www.bonfacemunyoki.com)
Chief Emacs Mchochezi / Twitter: @BonfaceKilz
GPG key = D4F09EB110177E03C28E2FE1F5BBAE1E0392253F

BonfaceKilz commented:
zsloan <notifications@github.com> writes:

> I think I managed to solve the issue of speed. It actually seems significantly faster than it was before using
> Scroller now (for pulling data out of the table).
>

Have you profiled things? Just IMO out of habit,
but it's a nice idea to profile things to be more
objective, kind of what Udit's been doing over on
e-mail.

> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.*

-- 
Bonface M. K. (https://www.bonfacemunyoki.com)
Chief Emacs Mchochezi / Twitter: @BonfaceKilz
GPG key = D4F09EB110177E03C28E2FE1F5BBAE1E0392253F

zsloan commented:
Ah, you mean measuring the run-time for the code (in this case javascript)?
I'll go ahead and do that.

In this case it's actually a very dramatic change (less than a second vs.
10+ seconds), but I agree that it's still good to have the specific times.

On Thu, Sep 24, 2020 at 8:52 PM BonfaceKilz <notifications@github.com>
wrote:

> zsloan <notifications@github.com> writes:
>
> > I think I managed to solve the issue of speed. It actually seems
> significantly faster than it was before using
> > Scroller now (for pulling data out of the table).
> >
>
> Have you profiled things? Just IMO out of habit,
> but it's a nice idea to profile things to be more
> objective, kind of what Udit's been doing over on
> e-mail.
>
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub, or unsubscribe.*
>
> --
> Bonface M. K. (https://www.bonfacemunyoki.com)
> Chief Emacs Mchochezi / Twitter: @BonfaceKilz
> GPG key = D4F09EB110177E03C28E2FE1F5BBAE1E0392253F
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/442#issuecomment-698676280>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGA37MTHK4NPPP74NRTSHPZV7ANCNFSM4RNPZ56A>
> .
>

zsloan committed:
Changed logic that defines the primary/other samplelists to be much more
efficient

* wqflask/wqflask/show_trait/show_trait.py - Add an "if" statement to
avoid unnecessarily running some code and changed a list to a set for
the purposes of searching if it contains an element

zsloan commented:
I was initially mistaking the speed with Scroller + my initial use of DataTables API for pre-Scroller, though it's still a bit faster than pre-Scroller
The page being tested on - http://gn2-zach.genenetwork.org/show_trait?trait_id=10443&dataset=HSNIH-PalmerPublish

With Scroller and first DataTables API use (that iterated through column cells) - 45.09458999999333 seconds
With Scroller and current DataTables API use (that just looks at individual columns and only does the "to JQuery" conversion on individual cells) - 0.12849000000278465 seconds
Before Scroller and switch to DataTables API - 0.17568499999470077 seconds

More noticeable is the amount of time it takes to block a range of samples by index (in this case blocking samples 1000-1100):

Before Scroller - 1.8479800000059186 seconds (also the page in general is sluggish here, with typing into text fields having lag)
With Scroller - 0.12387499999022111 seconds

zsloan committed:
Changed edit_data_change and block_by_index to use DataTables API

* wqflask/wqflask/static/new/javascript/show_trait.js - Changed
edit_data_change to use DataTables API to fetch the list of names,
values, and (where applicable) standard errors in a way that just looks
at those columns (previously it iterated through all rows).
block_by_index was also changed to look just at columns

zsloan committed:
Fixed issue where attribute columns' alignment was offset by 1

* wqflask/wqflask/static/new/javascript/initialize_show_trait_tables.js
- attribute cells' alignment and column names were wrong because they were referring to an attribute offset by 1 in the attribute list
-

robwwilliams commented:
Got the page to load correctly in Firefox in just under 10 seconds. That is
great.

No error messages.

On Fri, Sep 25, 2020 at 2:48 PM zsloan <notifications@github.com> wrote:

> I was initially mistaking the speed with Scroller + my initial use of
> DataTables API for pre-Scroller, though it's still a bit faster than
> pre-Scroller
> The page being tested on -
> http://gn2-zach.genenetwork.org/show_trait?trait_id=10443&dataset=HSNIH-PalmerPublish
>
> With Scroller and first DataTables API use (that iterated through column
> cells) - 45.09458999999333 seconds
> With Scroller and current DataTables API use (that just looks at
> individual columns and only does the "to JQuery" conversion on individual
> cells) - 0.12849000000278465 seconds
> Before Scroller and switch to DataTables API - 0.17568499999470077 seconds
>
> More noticeable is the amount of time it takes to block a range of samples
> by index (in this case blocking samples 1000-1100):
>
> Before Scroller - 1.8479800000059186 seconds (also the page in general is
> sluggish here, with typing into text fields having lag)
> With Scroller - 0.12387499999022111 seconds
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/442#issuecomment-699121581>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC55V7FJBONYKLFOIJLHTW3SHTXXLANCNFSM4RNPZ56A>
> .
>


-- 
Rob

Robert W. Williams, Ph.D.
Chair: Department of Genetics, Genomics and Informatics
University of Tennessee Health Science Center
71 S Manassas St, Memphis TN 38103-3310,  TSRB Room 405
Tele/Cell: 901 604 4752
EMAIL: rwilliams@uthsc.edu  or  labwilliams@gmail.com

zsloan committed:
There were two copies of the "submit_special" function in both show_trait.js and show_trait_mapping_tools.js. The latter now just calls the function from the former

* wqflask/wqflask/static/new/javascript/show_trait_mapping_tools.js - Everything in this file was previously inside its own function, which made its functions unable to access the functions from show_trait.js (and necessitated it having its own "copy" of the submit_special function

zsloan committed:
Added function that pulls input data out of table and inserts it as hidden input elements in the form (so form submission can work with Scroller)

* wqflask/wqflask/static/new/javascript/show_trait.js - Added function "get_table_contents_for_form_submit" that is called with submit_special (that is called for correlations and mapping from the trait page)

zsloan committed:
Converted transform functions to use DataTables API to work with scroller + fixed them to take into account blocked samples

* wqflask/wqflask/static/new/javascript/show_trait.js - Changed transform/normalize functions to use DataTables API to fetch and change values and also changed the way that code is structured to avoid a lot of duplicated code

zsloan committed:
Fixed issue that was causing SE column to not show up in the trait page table

* wqflask/wqflask/static/new/javascript/initialize_show_trait_tables.js - Apparently "true" in js_data was a boolean instead of a string, so the previous if statements weren't working properly

zsloan committed:
Added square root to the transform functions that are only active if there are no negative values

* wqflask/wqflask/templates/show_trait_transform_and_filter.html - made sqrt option contingent on no negative values

zsloan committed:
Changed order of JS imports to both make DataTables objects available to all other JS code and move all local JS code below third-party libraries

* wqflask/wqflask/templates/show_trait.html - Changed order of JS imports

zsloan committed:
Simplified if statements

* wqflask/wqflask/static/new/javascript/initialize_show_trait_tables.js - simplified if statements to just say if(js_data.se_exists) instead of if(js_data.se_exists === true)

zsloan committed:
Changed the way tranform functions work

* wqflask/wqflask/static/new/javascript/show_trait.js
- Changed transform options to only apply to original values, with the exception of the "invert" transform and rows where the value was set to "x"
- Only run "check_if_zero_to_one_vals" once on page load instead of running it every time a transform is run
- Changed Transform text to reflect when there's a combination of Invert and some other transform
- Updated Reset function to use DataTables API

zsloan committed:
Replaced "+" with "and" in transform text

* wqflask/wqflask/static/new/javascript/show_trait.js - Replaced "+" with "and" when Invert and another transform are combined and capitalized "Transform"

zsloan committed:
Added right-alignment for trait_value_input text fields in sample table to CSS file instead of setting it as a style in initialize_show_trait_tables.js

BonfaceKilz commented on wqflask/wqflask/show_trait/show_trait.py:
```diff
@@ -531,21 +540,21 @@ def get_nearest_marker(this_trait, this_db):
 
     trait_table_width = 380
-    if sample_groups[0].se_exists():
+    if sample_groups[0].se_exists:
```
**Question**: Not sure about this, but should this be `se_exists()` since it's been called?

zsloan commented:
se_exists was originally a method in SampleList, but I changed it to a
property since there didn't seem to be any reason for it to be a method.

On Tue, Oct 13, 2020 at 8:05 PM BonfaceKilz <notifications@github.com>
wrote:

> *@BonfaceKilz* commented on this pull request.
> ------------------------------
>
> In wqflask/wqflask/show_trait/show_trait.py
> <https://github.com/genenetwork/genenetwork2/pull/442#discussion_r504341777>
> :
>
> >  def get_table_widths(sample_groups, has_num_cases=False):
>      stats_table_width = 250
>      if len(sample_groups) > 1:
>          stats_table_width = 450
>
>      trait_table_width = 380
> -    if sample_groups[0].se_exists():
> +    if sample_groups[0].se_exists:
>
> *Question*: Not sure about this, but should this be se_exists() since
> it's been called?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/442#pullrequestreview-507926924>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGHF56CXQ7DFPLR2FBLSKT2NRANCNFSM4RNPZ56A>
> .
>

zsloan committed:
Merge branch 'testing' into scroller_testing

zsloan commented:
I think this is ready to merge; everything seems to work okay.

zsloan commented:
Actually, I just remembered that, before this is pulled to production, Scroller needs to be installed in GUIX (it's currently using the CDN link).

BonfaceKilz commented:
zsloan <notifications@github.com> writes:

> Actually, I just remembered that, before this is
> pulled to production, Scroller needs to be
> installed in GUIX (it's currently using the CDN
> link).

No need to worry about that. It's already
packaged. To use the js scroller from guix, you'd
have to do something like:

For the css(see
909029dea8ff71461120f8686825166c7cf7eb8b and
5774e985efa64c73f2deb5d0ace7b853d5fcf43b):

--8<---------------cut here---------------start------------->8---
<link rel="stylesheet" type="text/css" href="{{ url_for('js', filename='DataTablesExtensions/scroller/css/scroller.dataTables.min.css') }}">
--8<---------------cut here---------------end--------------->8---

And for JS:

--8<---------------cut here---------------start------------->8---
<script language="javascript" type="text/javascript" src="{{ url_for('js', filename='DataTablesExtensions/scroller/js/scroller.dataTables.min.js') }}"></script>    
--8<---------------cut here---------------end--------------->8---

Note that we use v2.0.1; I could update that if
necessary.

-- 
Bonface M. K. <https://www.bonfacemunyoki.com>
Chief Emacs Bazu / Rieng ya software sare
Mchochezi of: <https://upbookclub.com> / Twitter: @BonfaceKilz
GPG Key: D4F09EB110177E03C28E2FE1F5BBAE1E0392253F

zsloan commented:
Ah, nice. I'll go ahead and check that it works fine and merge if it does.

On Mon, Oct 19, 2020, 6:34 PM BonfaceKilz <notifications@github.com> wrote:

> zsloan <notifications@github.com> writes:
>
> > Actually, I just remembered that, before this is
> > pulled to production, Scroller needs to be
> > installed in GUIX (it's currently using the CDN
> > link).
>
> No need to worry about that. It's already
> packaged. To use the js scroller from guix, you'd
> have to do something like:
>
> For the css(see
> 909029dea8ff71461120f8686825166c7cf7eb8b and
> 5774e985efa64c73f2deb5d0ace7b853d5fcf43b):
>
> --8<---------------cut here---------------start------------->8---
> <link rel="stylesheet" type="text/css" href="{{ url_for('js',
> filename='DataTablesExtensions/scroller/css/scroller.dataTables.min.css')
> }}">
> --8<---------------cut here---------------end--------------->8---
>
> And for JS:
>
> --8<---------------cut here---------------start------------->8---
> <script language="javascript" type="text/javascript" src="{{ url_for('js',
> filename='DataTablesExtensions/scroller/js/scroller.dataTables.min.js')
> }}"></script>
> --8<---------------cut here---------------end--------------->8---
>
> Note that we use v2.0.1; I could update that if
> necessary.
>
> --
> Bonface M. K. <https://www.bonfacemunyoki.com>
> Chief Emacs Bazu / Rieng ya software sare
> Mchochezi of: <https://upbookclub.com> / Twitter: @BonfaceKilz
> GPG Key: D4F09EB110177E03C28E2FE1F5BBAE1E0392253F
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/442#issuecomment-712498381>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGGJAQZBBXDT3HAHTTLSLTEJTANCNFSM4RNPZ56A>
> .
>

BonfaceKilz commented:
zsloan <notifications@github.com> writes:

> Ah, nice. I'll go ahead and check that it works fine and merge if it does.

Cool :)

-- 
Bonface M. K. <https://www.bonfacemunyoki.com>
Chief Emacs Bazu / Rieng ya software sare
Mchochezi of: <https://upbookclub.com> / Twitter: @BonfaceKilz
GPG Key: D4F09EB110177E03C28E2FE1F5BBAE1E0392253F
